django/contrib/postgres/search.py

```diff
@@ -240,6 +242,65 @@ def as_sql(self, compiler, connection, function=None, template=None):
         return sql, extra_params + params
 
 
+class SearchHeadline(Func):
+    function = 'ts_headline'
+    output_field = TextField()
+    options = {}
+    VALID_OPTIONS = (
+        'StartSel',
+        'StopSel',
+        'MaxWords',
+        'MinWords',
+        'ShortWord',
+        'HighlightAll',
+        'MaxFragments',
+        'FragmentDelimiter',
+    )
```

Comment:
Should be a `frozenset` since it's only used for containment. From that point you use `set(options).difference(self.VALID_OPTIONS)` to determine `invalid_options`.